scripts/sign_encrypt.py

```diff
@@ -173,6 +173,17 @@ def main():
     with open(args.inf, 'rb') as f:
         img = f.read()
 
+    if sys.version_info[0] < 3:
+        tmp_encode=sys.getdefaultencoding()
+        reload(sys)
+        sys.setdefaultencoding('gbk')
```

Comment:
What is the purpose of this?

Reply:
If use python2 to excute sign_encrypt.py script and the first byte is upper than 0x80 in TA binary(such as the first byte is 0xb7)，calling struct.unpack("IIIIHH",xxx), it will occur following error:
`UnicodeDecodeError: ‘ascii’ codec can’t decode byte 0xb7 in position 18: ordinal not in range(128)`

The reason is that python2 used UTF-8 coding by default. So I save the value and change the coding format. After calling struct.unpack , restore the value of coding format.

I have already tested the script successfully with both python 3.6.4 and python 2.7.18.

Reply:
We don't support python 2 anymore as you can clearly see from the shebang line (`#!/usr/bin/env python3`), so please remove this.

Reply:
I agree with @jforissier: Please remove the check for python major version.